### PR TITLE
Add support output directory for result

### PR DESCRIPTION
**Pull request by danghai**
I make PR to add support output directory for result. If the command has `--output_dir` or `-o` option, it is going to create the output directory to store result. And the default is working directory.

### CODE REVIEW

File: skt/executable.py [hunk 1]
```diff
@@ -564,6 +564,12 @@ def setup_parser():
         type=str,
         help="Path to work dir"
     )
+    parser.add_argument(
+        "-o",
+        "--output_dir",
```
Comment (veruu) [inline skt/executable.py]: The command line options should use `-` instead of `_`

File: skt/executable.py [hunk 2]
```diff
@@ -564,6 +564,12 @@ def setup_parser():
...
+        type=str,
+        help="Path to output dir"
```
Comment (veruu) [inline skt/executable.py]: Can you use the full "directory" word here?

File: skt/executable.py [hunk 3]
```diff
@@ -966,8 +972,13 @@ def load_config(args):
             cfg['runner'][1]['blacklist']
         )
 
-    # Preparation for output directory option
-    if os.access(cfg.get('workdir'), os.W_OK | os.X_OK):
+    # Create and get an absolute path for the output directory
+    if cfg.get('output_dir'):
+        try:
+            os.mkdir(full_path(cfg.get('output_dir')))
```
Comment (veruu) [inline skt/executable.py]: Can you assign the full path to the output directory first and work with that? That way we don't need to convert it with every use.

**Comment by danghai** [comment]
@veruu thanks review. I have updated it.  Could you review it, and my commit message? It relates to issue #296 But I am not sure about what `input` for directory result is? Thanks

File: README.md [hunk 4]
```diff
@@ -131,18 +131,18 @@ written to the `<JUNIT_DIR>` directory.
...
+    skt --rc <SKTRC> --state --workdir <WORKDIR> --output-dir <OUTDIR> -vv \
         merge --baserepo <REPO_URL> --ref <REPO_REF>
 
 Here `<SKTRC>` would be the configuration file to retrieve the configuration
 and the state from, and store the updated state in. `<WORKDIR>` would be the
-directory to clone and checkout the kernel repo to, `<REPO_URL>` would be the
-source kernel Git repo URL, and `<REPO_REF>` would be the reference to
-checkout.
+directory to clone and checkout the kernel repo to, `<OUTDIR>` would be
+directory to store output files creation, `<REPO_URL>` would be the source
```
Comment (veruu) [inline README.md]: the `creation` word doesn't belong here

**Comment by veruu** [comment]
@danghai don't worry about mentioning the issue. It's a bit more complicated and you'd have to read the full discussion in the meta/pull6. We'd like to make skt more modular and get rid of workdir and potentially state as well.

**Comment by danghai** [comment]
@veruu okay. I am going to read it.

**Comment by danghai** [comment]
@veruu do I need to fix anything in this patch?

**Comment by veruu** [comment]
No, I think the pull is fine now that it's rebased. Will merge it soon, thanks!

**Pull request merged**